### DISCUSSION TITLE
foonathan_memory_vendor FORCE_BUILD option

### DISCUSSION THREAD

**Discussion by xavisolesoft** [eProsima/Fast-DDS#4528] [Ideas]
**Proposal**
I have a feature proposal for `foonathan_memory_vendor` package build:

My proposal consist on create a cmake option to be able to force the build even when ros is installed in the system path.
We need this option due we are using a different compiler from the system default one, so we need to force the build with our compiler.

**Proposed implementation**
I have an implementation proposal (I copy the snapshot because I have no permissions to create a branch in `foonathan_memory_vendor`):
![image](https://github.com/eProsima/Fast-DDS/assets/20727166/c3e22c7f-6141-4b6b-b410-2ddc9413e952)

**How to use it**
With the proposed implementation, force build can be enabled by passing the next argument to cmake in the command line:
```bash
-DFORCE_BUILD=ON
```

**Comment by JesusPoderoso** [comment]
Hi @xavisolesoft, thanks for using Fast DDS, and thanks for your proposal
Would you mind opening a Pull Request to follow up on it?
The procedure would be:
1. [Fork the vendor repo](https://github.com/eProsima/foonathan_memory_vendor/fork)
2. Implement the proposal on that fork
3. [Open the PR](https://github.com/eProsima/foonathan_memory_vendor/compare) from your fork to our repo 

Thanks in advance!

**Reply by xavisolesoft** [reply]
Thanks.

I created this PR as suggested:
https://github.com/eProsima/foonathan_memory_vendor/pull/69
